libcontainer/rootfs_linux.go

```diff
@@ -718,18 +747,12 @@ func checkProcMount(rootfs, dest, source string) error {
 		return nil
 	}
 	if path == "." {
-		// an empty source is pasted on restore
-		if source == "" {
-			return nil
-		}
 		// only allow a mount on-top of proc if it's source is "proc"
-		isproc, err := isProc(source)
+		st, err := m.srcStatfs()
 		if err != nil {
 			return err
 		}
-		// pass if the mount is happening on top of /proc and the source of
-		// the mount is a proc filesystem
-		if isproc {
+		if st.Type == unix.PROC_SUPER_MAGIC {
```

Comment:
Yes, we need to change this code to use `srcStatfs()` as part of this commit, and there are no other users of `isProc` so removing it makes the code easier to read. Otherwise we'd need to change the signature of a function we remove in a later commit.